tests/garage/tf/policies/test_gaussian_mlp_policy_with_model.py

```diff
@@ -0,0 +1,173 @@
+        log_prob2 = self.sess.run(  # result
+            log_prob_sym2,
+            feed_dict={policy3.model.networks['default'].input: obs})
+
+        assert log_prob1 == log_prob2
+
+        ############################################################
+        # entropy_sym
+        # * Existing way *
+        entropy_sym1 = policy1.distribution.entropy_sym(
+            dist1_sym, name='entropy_sym1')
+        entropy_func = tensor_utils.compile_function([obs_ph], entropy_sym1)
+        entropy1 = entropy_func(obs)  # result
+
+        # * New way *
+        # equvaient to garage.tf.distribution.entropy_sym(dist)
+        entropy_sym2 = policy3.model.networks['default'].dist.entropy()
+        entropy2 = self.sess.run(  # result
+            entropy_sym2,
+            feed_dict={policy3.model.networks['default'].input: obs})
+
+        assert entropy1 == entropy2
+
+        ############################################################
+        # likelihood_ratio_sym
+        # * Existing way *
+        likelihood_ratio_sym1 = policy1.distribution.likelihood_ratio_sym(
+            obs_ph, dist1_sym, dist2_sym, name='li_ratio_sym1')
+        likelihood_ratio_func = tensor_utils.compile_function(
+            [obs_ph], likelihood_ratio_sym1)
+        likelihood_ratio1 = likelihood_ratio_func(obs)
+
+        # * New way *
+        with tf.name_scope('li_ratio_sym2'):
+            likelihood_ratio_sym2 = policy4.likelihood_ratio_sym(
+                obs_ph, policy3.model.networks['default'].dist)
+            likelihood_ratio2 = self.sess.run(  # result
+                likelihood_ratio_sym2,
+                feed_dict={
+                    policy3.model.networks['default'].input: obs,
+                    policy4.model.networks['default'].input: obs,
+                    obs_ph: obs
+                })
+
+        assert likelihood_ratio1 == likelihood_ratio2
+
+    def test_guassian_mlp_policy_pickle(self):
+        box_env = TfEnv(DummyBoxEnv())
+        data = np.ones((3, 1))
+        with tf.Session(graph=tf.Graph()) as sess:
+            policy = GaussianMLPPolicyWithModel2(
+                env_spec=box_env, init_std=1.1)
+            # model is built in GaussianMLPPolicyWithModel2.__init__
+            outputs = sess.run(
+                policy.model.networks['default'].sample,
+                feed_dict={policy.model.networks['default'].input: data})
+            p = pickle.dumps(policy)
+
+        with tf.Session(graph=tf.Graph()) as sess:
+            policy_pickled = pickle.loads(p)
+            # After pickle, we need to build the model
+            # e.g. by policy.dist_info_sym
+            input_ph = box_env.observation_space.new_tensor_variable(
+                extra_dims=1, name='input_ph')
+            policy_pickled.dist_info_sym(
+                input_ph, tfp.distributions.MultivariateNormalDiag)
+
+            outputs2 = sess.run(
+                policy_pickled.model.networks['default'].sample,
```

Comment:
seems like this is ripe for a helper function on `model`, but that can be for another PR